File: .ProgramingBasics/AdvancedLoops/10.CheckPrime/src/CheckPrime.java
import java.util.Scanner;

public class CheckPrime {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        boolean isPrime = true;

        if (n < 2) {
            isPrime = false;
        }
        else {
            for (int i = 2; i <= Math.sqrt(n); i++) {
                if (n % i == 0) {
                    isPrime = false;
                    break;
                }
            }
        }

        System.out.println(isPrime ? "Prime" : "Not prime");
    }
}

File: ProgramingBasics/AdvancedLoops/10.CheckPrime/10.CheckPrime/CheckPrime.cs
using System;

namespace _10.CheckPrime
{
    public class CheckPrime
    {
        public static void Main(string[] args)
        {
            var n = int.Parse(Console.ReadLine());
            var isPrime = true;

            if (n < 2)
            {
                isPrime = false;
            }
            else
            {
                for (int i = 2; i <= Math.Sqrt(n); i++)
                {
                    if (n % i == 0)
                    {
                        isPrime = false;
                        break;
                    }
                }
            }

            Console.WriteLine(isPrime ? "Prime" : "Not prime");
        }
    }
}